dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/DefaultCodec.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.remoting.transport.netty4;
+
+import org.apache.dubbo.remoting.Channel;
+import org.apache.dubbo.remoting.Codec2;
+import org.apache.dubbo.remoting.buffer.ChannelBuffer;
+
+import java.io.IOException;
+
+public class DefaultCodec implements Codec2 {
```

Comment:
Why add an empty implementation？

Reply:
super class of AbstractPortUnificationServer need a codec implemention, or throw Exception if it can't find. This implemention is to make PortUnificationServer work normal.  

<img width="1110" alt="image" src="https://user-images.githubusercontent.com/47651330/179340317-a02ce626-85fb-4bf2-8727-d7b27f76883f.png">

Reply:
And for now codec of triple is alse an empty implemention. Codec is to make encoding and decoding independent from specific NIO framework. And in future, to make triple protocol independent from netty4.(Maybe a new PortUnificationServer implementing with other NIO framework) we will need to implement a codec for triple.